pkg/tests/tasks/security/certmanager/istio_csr_test.go

```diff
@@ -0,0 +1,121 @@
+package certmanager
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/maistra/maistra-test-tool/pkg/app"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/common"
+	"github.com/maistra/maistra-test-tool/pkg/util/curl"
+	"github.com/maistra/maistra-test-tool/pkg/util/env"
+	"github.com/maistra/maistra-test-tool/pkg/util/helm"
+	"github.com/maistra/maistra-test-tool/pkg/util/istio"
+	"github.com/maistra/maistra-test-tool/pkg/util/ns"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/template"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+	"github.com/maistra/maistra-test-tool/pkg/util/version"
+)
+
+func TestIstioCsr(t *testing.T) {
+	test.NewTest(t).Id("T38").Groups(test.Full, test.ARM, test.InterOp).Run(func(t test.TestHelper) {
+		smcpVer := env.GetSMCPVersion()
+		if smcpVer.LessThan(version.SMCP_2_4) {
+			t.Skip("istio-csr is not supported in SMCP older than v2.4")
+		}
+
+		meshValues := map[string]string{
+			"Name":    smcpName,
+			"MeshNs":  meshNamespace,
+			"Member":  ns.Foo,
+			"Version": smcpVer.String(),
+		}
+		istioCsrValues := map[string]string{
+			"MeshNs":   meshNamespace,
+			"Revision": smcpName,
+		}
+
+		t.Cleanup(func() {
+			helm.Namespace(meshNamespace).Release("istio-csr").Uninstall(t)
+			oc.DeleteFromTemplate(t, meshNamespace, serviceMeshIstioCsrTmpl, meshValues)
+			oc.DeleteFromString(t, meshNamespace, istioCA)
+			oc.DeleteSecret(t, meshNamespace, "istiod-tls")
+			oc.DeleteSecret(t, meshNamespace, "istio-ca")
+			oc.RecreateNamespace(t, ns.Foo)
+		})
+
+		t.LogStep("Uninstall existing SMCP")
+		oc.RecreateNamespace(t, meshNamespace)
+
+		t.LogStep("Create intermediate certificate for Istio")
+		retry.UntilSuccess(t, func(t test.TestHelper) {
```

Comment:
I wrapped it in `retry.UntilSuccess`, because I remember that Marko encountered failures during creation of certificates and issuers. cert-manager was rejecting requests for some reason.

Removed, thanks.